osquery/logger/plugins/aws_firehose.cpp

```diff
@@ -105,20 +105,17 @@ Status FirehoseLogForwarder::setUp() {
   }
 
   // Make sure we can connect to designated stream
-  Aws::Firehose::Model::ListDeliveryStreamsRequest r;
-  auto result = client_->ListDeliveryStreams(r).GetResult();
-  std::vector<std::string> stream_names = result.GetDeliveryStreamNames();
-
-  if (std::find(stream_names.begin(),
-                stream_names.end(),
-                FLAGS_aws_firehose_stream) == stream_names.end()) {
-    return Status(1,
-                  "Could not find Firehose delivery stream: " +
-                      FLAGS_aws_firehose_stream);
+  Aws::Firehose::Model::DescribeDeliveryStreamRequest r;
+  r.SetDeliveryStreamName(FLAGS_aws_firehose_stream);
+  auto result = client_->DescribeDeliveryStream(r).GetResult();
+  auto description = result.GetDeliveryStreamDescription();
+  auto stream_name = description.GetDeliveryStreamName();
+  if (stream_name.empty()){
```

Comment:
For sure, I'm not certain that there is a problem, but I'm also not certain that there's not. I feel like the previous author, @zwass, probably was certain that the code was safe and we should have similar confidence before proceeding.